plugins/inputs/gnmi/handler.go

```diff
@@ -26,20 +29,30 @@ type handler struct {
 	tagsubs            []TagSubscription
 	maxMsgSize         int
 	emptyNameWarnShown bool
+	jnprExtension      bool
 	tagStore           *tagStore
 	trace              bool
 	log                telegraf.Logger
 }
 
-func newHandler(addr string, aliases map[string]string, subs []TagSubscription, maxsize int, l telegraf.Logger, trace bool) *handler {
+func newHandler(addr string, aliases map[string]string, subs []TagSubscription, maxsize int, l telegraf.Logger, addConf AdditionalConf) *handler {
+	// parse vendor specific and set vendor specific variables
+	// Current supported vendor specific option is jnpr_extension
+	jnprext := false
+	for _, val := range addConf.VendorExt {
+		if val == "jnpr_extension" {
+			jnprext = true
+		}
+	}
```

Comment:
Rather than going through the vendor extensions here, why not pass them down even further to `handleSubscribeResponseUpdate`?

Reply:
Yes. I will change this. Using a config struct instead.